core/blockchain_test.go

```diff
@@ -1810,3 +1699,230 @@ func TestPrunedImportSide(t *testing.T) {
 	testSideImport(t, 1, 10)
 	testSideImport(t, 1, -10)
 }
+
+func TestInsertKnownHeaders(t *testing.T)      { testInsertKnownChainData(t, "headers") }
```

Comment:
Great that you added testcases! Could we also have the following:

1. Two competing chains, where `A` is longer, but `B` has higher td, 
2. Both are imported (known). 
3. We feed chain `A` again into the importer to see what happens. Maybe in combination with a setHead where we rollback to some common ancestor

Reply:
Added